### PR TITLE
feat: support databend module

### PR DESCRIPTION
**Pull request by hantmac**
## What does this PR do?
Support new module: [databend](https://github.com/datafuselabs/databend)

### CODE REVIEW

**Review by mdelapenya — CHANGES_REQUESTED** [review]
@hantmac thanks for adding this new module, much appreciated! The go code looks good, although we need to add an the metadata for the module to be included in the CI and in the docs.

Could you please run the module generator code for it? https://golang.testcontainers.org/modules/#creating-a-new-module

Thanks!

**Comment by mdelapenya** [comment]
@hantmac I went ahead and added the scaffolding. Could you please enrich the docs with the new functional options so that users find them in our docs site 🙏 ?

Thanks in advance

File: modules/databend/databend.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+type DatabendContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+func WithDefaultCredentials() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		username := req.Env["QUERY_DEFAULT_USER"]
```
Comment (mdelapenya) [inline modules/databend/databend.go:29]: I think you want to transfer those values to the proper env var, right? Otherwise, those vars won't take effect

Comment (hantmac) [reply modules/databend/databend.go:29]: Yes, I want to make sure the ENV exists. I will fix it.

**Comment by hantmac** [comment]
> @hantmac I went ahead and added the scaffolding. Could you please enrich the docs with the new functional options so that users find them in our docs site 🙏 ?
> 
> Thanks in advance

HI @mdelapenya , would you like to explain what is new functional options or where should I found it?

**Comment by hantmac** [comment]
@mdelapenya HI bro, I saw you have add the docs and some CI workflow information, thanks very much! Is there some other things I need to do?

File: docs/modules/databend.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+When starting the Databend container, you can pass options in a variadic way to configure it.
+
+#### Image
+
+If you need to set a different Databend Docker image, you can set a valid Docker image as the second argument in the `Run` function.
+E.g. `Run(context.Background(), "datafuselabs/databend:v1.2.615")`.
+
+{% include "../features/common_functional_options.md" %}
+
```
Comment (mdelapenya) [inline docs/modules/databend.md:52]: @hantmac it would be convenient to document the new functional options for databend here at this point. Please use other modules as reference (e.g. MySQL: https://golang.testcontainers.org/modules/mysql/#set-username-password-and-database-name)

In this PR, the module is exposing two options: https://github.com/testcontainers/testcontainers-go/pull/2779/files#diff-05e51be59eadcbdc57a90dccc140946369a930b7bec7e9e987096ebed8beb9beR127, and two container methods (see https://github.com/testcontainers/testcontainers-go/pull/2779/files#diff-05e51be59eadcbdc57a90dccc140946369a930b7bec7e9e987096ebed8beb9beR103). Please add the container methods right after L55: https://github.com/testcontainers/testcontainers-go/pull/2779/files#diff-b3c18d18db5bfc68feda99c0eb84554cb77251863f0f2a5237133eced8fbc72eR55

Comment (mdelapenya) [reply docs/modules/databend.md:52]: BTW the render URL of the docs for this PR is here: https://deploy-preview-2779--testcontainers-go.netlify.app/modules/databend/

Comment (hantmac) [reply docs/modules/databend.md:52]: Thanks for your help! I fixed the dcs.

File: modules/databend/databend.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+// DatabendContainer represents the Databend container type used in the module
+type DatabendContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+func WithDefaultCredentials() testcontainers.CustomizeRequestOption {
```
Comment (mdelapenya) [inline modules/databend/databend.go:27]: One last thing before running the CI: as this function is exposed in the API, I think we need to add it in the docs, unless it's private and not intended to be part of the public API.

Wdyt about that? Do we want users to define default credentials, or we should keep it internal to the module as an implementation detail?

Comment (hantmac) [reply modules/databend/databend.go:27]: Sorry, I have not seen the `WithDefaultCredentials ` function docs in mysql.md,  maybe we should keep it internal to the module.

Comment (mdelapenya) [reply modules/databend/databend.go:27]: Yeah, each module comes with its own set of configuration options, and its own set of internals. I agree, this option seems internal to the databend module

Comment (stevenh) [reply modules/databend/databend.go:27]: This cant be private as the user needs to call pass it to `Run`

File: modules/databend/databend.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	if extraArgs != "" {
+		extraArgs = "?" + extraArgs
+	}
+
+	// databend://databend:databend@localhost:8000/default?sslmode=disable
+	connectionString := fmt.Sprintf("databend://%s:%s@%s:%s/%s%s", c.username, c.password, host, containerPort.Port(), c.database, extraArgs)
+	return connectionString, nil
+}
+
+func WithUsername(username string) testcontainers.CustomizeRequestOption {
```
Comment (stevenh) [inline modules/databend/databend.go:127]: suggestion: can be document all public methods please. While they seem self explanatory it still helps the developer experience.
```suggestion
// WithUsername is [Run] option that configures the default query user by setting
// the `QUERY_DEFAULT_USER` container environment variable.
func WithUsername(username string) testcontainers.CustomizeRequestOption {
```

File: modules/databend/databend_test.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	ctx := context.Background()
+
+	ctr, err := databend.Run(ctx,
+		"datafuselabs/databend:v1.2.615",
+		databend.WithUsername("databend"))
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	// perform assertions
+	connectionString, _ := ctr.ConnectionString(ctx)
```
Comment (stevenh) [inline modules/databend/databend_test.go:65]: bug: missing error check
```suggestion
	connectionString, err := ctr.ConnectionString(ctx)
	require.NoError(t, err)
```

File: modules/databend/examples_test.go [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	// }
+
+	state, err := databendContainer.State(ctx)
+	if err != nil {
+		log.Printf("failed to get container state: %s", err)
+		return
+	}
+
+	fmt.Println(state.Running)
```
Comment (stevenh) [inline modules/databend/examples_test.go:38]: bug: missing output definition to ensure the example validates

File: modules/databend/examples_test.go [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+		log.Printf("failed to query: %s", err)
+		return
+	}
+	err = row.Scan(&i)
+	if err != nil {
+		log.Printf("failed to scan row: %s", err)
+		return
+	}
+
+	fmt.Println(i)
```
Comment (stevenh) [inline modules/databend/examples_test.go:88]: bug: Missing Output definition for validation of the example.

File: modulegen/_template/ci.yml.tmpl [hunk 8]
```diff
@@ -129,7 +129,7 @@ jobs:
     runs-on: ubuntu-latest 
     steps:
       - name: Check out code into the Go module directory
-        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.1
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (stevenh) [inline modulegen/_template/ci.yml.tmpl:132]: suggestion: This seems to be an unrelated change not mentioned in PR, can we remove or document the reason?

File: docs/modules/databend.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+<!--/codeinclude-->
+
+## Module Reference
+
+### Run function
+
+- Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+!!!info
+    The `RunContainer(ctx, opts...)` function is deprecated and will be removed in the next major release of _Testcontainers for Go_.
```
Comment (stevenh) [inline docs/modules/databend.md:30]: bug: There is no such method, might be an old template?

**Review by stevenh — CHANGES_REQUESTED** [review]
Thanks for working on this, a few bugs to fix and a few questions, where I'd like some clarification.

File: modules/databend/databend.go (hunk 3 above)
Comment (hantmac) [reply modules/databend/databend.go:27]: @stevenh You are right. IMHO, this method is redundant, I should delete this method.

File: docs/modules/databend.md (hunk 9 above)
Comment (hantmac) [reply docs/modules/databend.md:30]: Yes, it is generated by the template.

File: modulegen/_template/ci.yml.tmpl (hunk 8 above)
Comment (hantmac) [reply modulegen/_template/ci.yml.tmpl:132]: cc @mdelapenya , how we document here?

File: modules/databend/databend.go [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+	opts = append(opts, WithDefaultCredentials())
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, err
+		}
+	}
+
+	username, ok := req.Env["QUERY_DEFAULT_USER"]
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go]: Fixed.

File: modules/databend/databend.go [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *DatabendContainer
+	if container != nil {
+		c = &DatabendContainer{
+			Container: container,
+			password:  password,
+			username:  username,
+			database:  defaultDatabaseName,
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:78]: I add a `WithDatabase` method using defined a type to satisfy the testcontainers.ContainerCustomizer interface.

File: modules/databend/examples_test.go (hunk 6 above)
Comment (hantmac) [reply modules/databend/examples_test.go:40]: Fixed.

File: docs/modules/databend.md (hunk 9 above)
Comment (hantmac) [reply docs/modules/databend.md]: Have removed.

File: modules/databend/databend.go [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+// Run creates an instance of the Databend container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DatabendContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{"8000/tcp"},
+		Env: map[string]string{
+			"QUERY_DEFAULT_USER":     defaultUser,
+			"QUERY_DEFAULT_PASSWORD": defaultPassword,
+		},
+		WaitingFor: wait.ForLog("port: 8000  Databend Query Server"),
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go]: Thanks!

File: modulegen/_template/ci.yml.tmpl (hunk 8 above)
Comment (mdelapenya) [reply modulegen/_template/ci.yml.tmpl:132]: This was added by me, so I think we can either roll it back, or simply include it in the description of the PR.

File: modules/databend/databend.go [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.Env["QUERY_DEFAULT_PASSWORD"] = password
+		return nil
+	}
+}
+
+// WithDatabase sets the name of the database to use.
+func WithDatabase(database string) DatabendOption {
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:137]: We also need to add it to the docs site.

File: modules/databend/examples_test.go [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+	"log"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/databend"
+)
+
+func TestExampleRun(t *testing.T) {
```
Comment (mdelapenya) [inline modules/databend/examples_test.go:16]: We do not add testing code in the testable examples file, as they will land in the pkg.go.dev site as code snippets. Please see https://go.dev/blog/examples as reference

File: modules/databend/examples_test.go (hunk 6 above)
Comment (mdelapenya) [reply modules/databend/examples_test.go:40]: We need here something like this for the testable example to be runnable by the test runner:
```suggestion
	fmt.Println(state.Running)

	// Output:
	// true
```

File: modules/databend/examples_test.go [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		log.Printf("failed to get container state: %s", err)
+		return
+	}
+
+	fmt.Println(state.Running)
+	assert.Equal(t, true, state.Running)
+}
+
+func TestExampleRun_connect(t *testing.T) {
```
Comment (mdelapenya) [inline modules/databend/examples_test.go:44]: Same here

File: modules/databend/databend.go (hunk 4 above)
Comment (mdelapenya) [reply modules/databend/databend.go:122]: Let's add the function comments too, as they will land into the pkg.go.dev site.

It could be seen as redundant because we do have our own docs site (golang.testcontainers.org), but in the latter we have control on the presentation.

File: modules/databend/databend.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:78]: You probably need a way to transfer the default options (db name) to the container request (as an env var??) so that the databend container uses that db name.

**Review by mdelapenya — CHANGES_REQUESTED** [review]
We are close 😊 Just a few comments from my side plus Steven's.

Thanks for your patience during the review 🙏

File: modules/databend/databend.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:78]: That' why I define a custom type, there has no DB env in databend container. 😢

File: modules/databend/databend.go (hunk 13 above)
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:137]: The db name will add to DSN when call `ConnectionString`, so that user can use the db.

File: modules/databend/examples_test.go (hunk 6 above)
Comment (hantmac) [reply modules/databend/examples_test.go:37]: Got it and fixed.

File: modules/databend/examples_test.go (hunk 14 above)
Comment (hantmac) [reply modules/databend/examples_test.go]: Got it.

File: modules/databend/examples_test.go (hunk 15 above)
Comment (hantmac) [reply modules/databend/examples_test.go]: Yep, fixed.

File: modules/databend/examples_test.go (hunk 7 above)
Comment (hantmac) [reply modules/databend/examples_test.go:91]: Thanks, fixed.

**Review by stevenh — CHANGES_REQUESTED** [review]
I've unresolved a few items, which hadn't been fixed, not sure if something didn't get committed.

Also looks like there was bugs introduced, unless I'm missing something?

File: modules/databend/databend.go [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{"8000/tcp"},
+		Env: map[string]string{
+			"QUERY_DEFAULT_USER":     defaultUser,
+			"QUERY_DEFAULT_PASSWORD": defaultPassword,
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog("port: 8000  Databend Query Server"),
+			wait.ForListeningPort("8000/tcp"),
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go]: Agree. Fixed.

File: modules/databend/examples_test.go [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/testcontainers/testcontainers-go/modules/databend"
+)
+
+func ExampleRun() {
+	ctx := context.Background()
+
+	databendContainer, err := databend.Run(ctx,
+		"datafuselabs/databend:v1.2.615",
+		databend.WithUsername("databend"),
+		databend.WithPassword("databend"),
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/examples_test.go]: Make sense. Fixed.

File: modules/databend/databend.go [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+	extraArgs := ""
+	if len(args) > 0 {
+		extraArgs = strings.Join(args, "&")
+	}
+	if extraArgs != "" {
+		extraArgs = "?" + extraArgs
+	}
+	if c.database == "" {
+		c.database = defaultDatabaseName
+	}
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:117]: User can use `c.ConnectionString` to switch different database which is the same as extra arg?

File: modules/databend/databend_test.go [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+func TestDatabendWithNonRootUserAndEmptyPassword(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := databend.Run(ctx,
+		"datafuselabs/databend:v1.2.615",
+		databend.WithUsername("databend"),
+		databend.WithPassword("databend"))
+	testcontainers.CleanupContainer(t, ctr)
+	require.Error(t, err)
+}
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend_test.go]: Thanks! fixed.

File: modules/databend/examples_test.go [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+	err = row.Scan(&i)
+	if err != nil {
+		log.Printf("failed to scan result: %s", err)
+		return
+	}
+
+	fmt.Println(i)
+
+	// Output:
+	// true
```
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/examples_test.go]: Fixed.

File: modules/databend/databend.go (hunk 13 above)
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:143]: The container doesn't need default DB ENV, but the connection dsn need a default one.

File: modules/databend/databend.go (hunk 18 above)
[the remark this answers is not in the document]
Comment (stevenh) [reply modules/databend/databend.go:117]: My understanding is that the user would use ConnectionString to enable them to connect to the container however they might have more than one database so passing that as a parameter into this method would enable that to work. Otherwise it they would have to create a new container or manually manipulate the result to do that.

It would be simple to have this method look for a database parameter in extra params and override the default to make that flow easy.

Does that make sense?

File: modules/databend/databend.go [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+// Run creates an instance of the Databend container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DatabendContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{"8000/tcp"},
+		Env: map[string]string{
+			"QUERY_DEFAULT_USER":     defaultUser,
+			"QUERY_DEFAULT_PASSWORD": defaultPassword,
+		},
+		WaitingFor: wait.ForAll(
```
Comment (stevenh) [inline modules/databend/databend.go:48]: suggestion: remove the ForAll as it's not needed

File: modules/databend/databend.go [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	extraArgs := ""
+	if len(args) > 0 {
+		extraArgs = strings.Join(args, "&")
+	}
+	if extraArgs != "" {
+		extraArgs = "?" + extraArgs
+	}
+	if c.database == "" {
```
Comment (stevenh) [inline modules/databend/databend.go:115]: suggestion: I would expect this be an error as the default value is not blank. That ensures the user knows they did something wrong instead of magically succeeding

**Review by stevenh — CHANGES_REQUESTED** [review]

File: modules/databend/databend.go (hunk 18 above)
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:117]: I think user can use `db.execute("use newDatabase")` to switch which is more convenient.  If add it in extra params, user also need to create a new connection string to connect.

File: modules/databend/databend.go (hunk 21 above)
Comment (hantmac) [reply modules/databend/databend.go]: Make sense. Fixed.

File: modules/databend/databend.go (hunk 22 above)
Comment (hantmac) [reply modules/databend/databend.go:113]: Agree. Fixed.

**Review by stevenh — APPROVED** [review]
Looks good to me will let @mdelapenya double check

File: modules/databend/databend.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:75]: Ok, understood. What I do not see is how the `WithDatabase` option is passing its value to the `container.database` field. In general, we usually do like this: https://github.com/testcontainers/testcontainers-go/blob/main/modules/redpanda/redpanda.go#L100-L109, defining a settings struct that will receive the configuration by the functional option, and then passing its values to the container.

File: modules/databend/databend.go (hunk 13 above)
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:141]: I'm not a databend expert, so I wonder how it creates a given database. In other words: as a user, if I set the database name with this functional option, what do I expect?
- the db is created in the databend container
- the container provides the right connection string to that db

**Comment by mdelapenya** [comment]
One thing that I'm seeing in the tests is that the `databend` SQL driver is not added (see https://github.com/testcontainers/testcontainers-go/actions/runs/10871947857/job/30189305158?pr=2779)

```
  2024/09/16 09:29:33 ⏳ Waiting for container id f43111bac8d5 image: datafuselabs/databend:v1.2.615. Waiting for: &{Port:8000/tcp timeout:<nil> PollInterval:100ms skipInternalCheck:false}
  2024/09/16 09:29:34 🔔 Container is ready: f43111bac8d5
      databend_test.go:31: 
          	Error Trace:	/home/runner/work/testcontainers-go/testcontainers-go/modules/databend/databend_test.go:31
          	Error:      	Received unexpected error:
          	            	sql: unknown driver "databend" (forgotten import?)
          	Test:       	TestDatabend
```

Could you verify that?

**Comment by hantmac** [comment]
> One thing that I'm seeing in the tests is that the `databend` SQL driver is not added (see https://github.com/testcontainers/testcontainers-go/actions/runs/10871947857/job/30189305158?pr=2779)
> 
> ```
>   2024/09/16 09:29:33 ⏳ Waiting for container id f43111bac8d5 image: datafuselabs/databend:v1.2.615. Waiting for: &{Port:8000/tcp timeout:<nil> PollInterval:100ms skipInternalCheck:false}
>   2024/09/16 09:29:34 🔔 Container is ready: f43111bac8d5
>       databend_test.go:31: 
>           	Error Trace:	/home/runner/work/testcontainers-go/testcontainers-go/modules/databend/databend_test.go:31
>           	Error:      	Received unexpected error:
>           	            	sql: unknown driver "databend" (forgotten import?)
>           	Test:       	TestDatabend
> ```
> 
> Could you verify that?

@mdelapenya Thanks! I forgotten import the databend-go driver. Fixed!

**Review by stevenh — CHANGES_REQUESTED** [review]
Just need the imports fixing into the correct blocks as identified by the linter

**Comment by hantmac** [comment]
> Just need the imports fixing into the correct blocks as identified by the linter

Thanks! @stevenh , fixed the ci lint.

File: modulegen/_template/ci.yml.tmpl (hunk 8 above)
Comment (mdelapenya) [reply modulegen/_template/ci.yml.tmpl:132]: The reason for this change is that the module generator was not updating the GH action version, that's why I used the right version. I can roll it back and submit a new PR.

File: modules/databend/databend.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:75]: I'm unresolving this comment because otherwise the database setting won't be applied to the container. @hantmac I can assist you and add it myself if needed 😊

[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:75]: The `WithDatabase` can't pass its value to the `container.database` directly, or though we pass it, it won't take effect until we execute `use newDatabase` or create another new connection dsn string to make a new connect.

File: modules/databend/databend.go (hunk 13 above)
[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:141]: @mdelapenya Hi, as a user, if you want to create a new database, you must execute `create database NEWDATABASE` and execute `use NEWDATABASE` to switch to it. So I think the `WithDatabase` function maybe not suitable for Databend?

[the remark this answers is not in the document]
Comment (stevenh) [reply modules/databend/databend.go:141]: Does that mean a hardcoded DB is always created?

[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:141]: Yep, the `default` dababase exists when `Databend` start.

[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:141]: > So I think the WithDatabase function maybe not suitable for Databend?
@hantmac then, I'd remove the withDatabase support and add it in a follow up: adding the option and creating the new database using the commands you posted above, as part of the container lifecycle hooks (postStarts) in order to create that database. Wdyt?

[the remark this answers is not in the document]
Comment (hantmac) [reply modules/databend/databend.go:141]: @mdelapenya I agree with you.

File: modules/databend/databend.go [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+		return "", err
+	}
+
+	extraArgs := ""
+	if len(args) > 0 {
+		extraArgs = strings.Join(args, "&")
+	}
+	if extraArgs != "" {
+		extraArgs = "?" + extraArgs
+	}
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply modules/databend/databend.go:112]: @hantmac could you also address this? 🙏

**Review by mdelapenya — CHANGES_REQUESTED** [review]
@hantmac let's remove all references to the `WithDatabase`, and once there. I think we are good to go.

**Comment by hantmac** [comment]
> @hantmac let's remove all references to the `WithDatabase`, and once there. I think we are good to go.

@mdelapenya Have removed the references to the `WithDatabase`, thanks for your patience.

File: modulegen/_template/ci.yml.tmpl (hunk 8 above)
Comment (mdelapenya) [reply modulegen/_template/ci.yml.tmpl:132]: @stevenh is right. I'll fix that in a separate PR. Sorry for the noise 🙏 
```suggestion
        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.1
```

**Comment by hantmac** [comment]
@mdelapenya It's strange the mssql CI test failed. q.q

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks for your hard work adding this new module. Very useful!

**Comment by mdelapenya** [comment]
> @mdelapenya It's strange the mssql CI test failed. q.q

The MSSQL failures have been resolved in #2786

**Comment by hantmac** [comment]
@mdelapenya Thanks for your help! BTW, I add the testcontainer-rs module in this pr https://github.com/testcontainers/testcontainers-rs-modules-community/pull/207, is there anybody who can help to review it?

**Comment by mdelapenya** [comment]
@hantmac could you please add this module to the modules catalog? https://testcontainers.com/modules/?language=go

There are instructions in [this repo](https://github.com/testcontainers/community-module-registry/) to elaborate the PR.

Regarding the Rust module, let me ping the community team on Slack

**Comment by hantmac** [comment]
> @hantmac could you please add this module to the modules catalog? https://testcontainers.com/modules/?language=go
> 
> There are instructions in [this repo](https://github.com/testcontainers/community-module-registry/) to elaborate the PR.
> 
> Regarding the Rust module, let me ping the community team on Slack

@mdelapenya Of course, I will add it. 

That's greate! Thank you!

**Comment by hantmac** [comment]
> There are instructions in [this repo](https://github.com/testcontainers/community-module-registry/) to elaborate the PR.

@mdelapenya Hi bro, I add the module to the modules catalog in this pr https://github.com/testcontainers/community-module-registry/pull/75
